src/app/recipe-detail-page/recipe-detail-page.component.ts

```diff
@@ -1,28 +1,315 @@
+    this.recipe_new.author = author;
+    this.checkInvalid();
+  }
+
+  getErrorMessageAuthor() {
+    this.color = "basic"
+    if(this.author.hasError('required')) {
+      return 'Please enter a author for your recipe!';
+    } else if(this.title.hasError('maxlength')) {
+      return 'Your authorname is too long. (Max length 255)'
+    } else{
+      return '';
+    }
+  }
+
+  inputDuration(event) {
+    const duration = event.target.value;
+    this.recipe_new.duration = duration;
+    this.checkInvalid();
+  }
+
+  getErrorMessageDuration() {
+    this.color = "basic";
+    if(this.duration.hasError('maxlength')) {
+      return 'Your duration is too long. Max 11 signs.';
+    } else {
+      return '';
+    }
+  }
+
+  inputNutritional(event) {
+    const nutritional = event.target.value;
+    this.recipe_new.nutritional = nutritional;
+    this.checkInvalid();
+  }
+
+  getErrorMessageNutritional() {
+    this.color = "basic";
+    if(this.nutritional.hasError('maxlength')) {
+      return 'Your nutritional values are too long. Max 255 signs.';
+    } else {
+      return '';
+    }
+  }
+
+  inputIngredient(event, ind: number) {
+    const ingredient = event.target.value;
+    this.recipe_new.addIngredient(ingredient, ind);
+    this.checkInvalid();
+  }
+
+  addIngredient() {
+    this.recipe_new.addIngredient("", -1);
+    this.highedit++;
+    this.highview++;
+    this.high = this.highedit;
+  }
+
+  deleteIngredient(index: number) {
+    this.recipe_new.deleteIngredient(index);
+    if(this.recipe_new.ingredient.length > 1) {
+      this.highedit--;
+      this.highview--;
+      this.high = this.highedit;
+    }
+  }
+
+  inputMaterial(event, ind: number) {
+    const material = event.target.value;
+    this.recipe_new.addMaterial(material, ind);
+    this.checkInvalid();
+  }
+
+  addMaterial() {
+    this.recipe_new.addMaterial("", -1);
+    this.highedit++;
+    this.highview++;
+    this.high = this.highedit;
+  }
+
+  deleteMaterial(index: number) {
+    this.recipe_new.deleteMaterial(index);
+    if(this.recipe_new.material.length > 1) {
+      this.highedit--;
+      this.highview--;
+      this.high = this.highedit;
+    }
+  }
+
+  inputStep(event, ind: number) {
+    const step = event.target.value;
+    this.recipe_new.addStep(step, ind);
+    this.checkInvalid();
+  }
+
+  addStep() {
+    this.recipe_new.addStep("", -1);
+    this.highedit++;
+    this.highview++;
+    this.high = this.highedit;
+  }
+
+  deleteStep(index: number) {
+    this.recipe_new.deleteStep(index);
+    if(this.recipe_new.step.length > 1) {
+      this.highedit--;
+      this.highview--;
+      this.high = this.highedit;
+    }
+  }
+
+  inputLink(event) {
+      const link = event.target.value;
+      this.recipe_new.link = link;
+      this.checkInvalid();
+    }
+
+  getErrorMessageLink() {
+    this.color = "basic";
+    if(this.nutritional.hasError('maxlength')) {
+      return 'Your link is too long. Max 255 signs.';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MyCookieBook/CookieBookFE/pullRequest?prid=7556402)
```suggestion
      return "Your link is too long. Max 255 signs.";
```